### ISSUE TITLE
Returning Incomplete Chunk

### ISSUE THREAD

**Issue by Qualcuno123** [inet-framework/inet#661]
Hello!
I'm doing a simulation in OMNeT++ 5.6.2 to simulate a network using INET library.
Everything is okay when I start the simulation, but after some time I get this error:

`Returning an incomplete chunk is not allowed according to the flags: 1 -- in module (inet::TcpGenericServerApp) Network.server1.app[17] (id=149), at t=5.9083416s, event #5999
`

I'll attach the .ini and .ned files of the simulation. I've neither created nor modified .cc files.
When I change the value of some parameters of the simulation this error may pop up later on in the simulation.
For example, looking at the .ned file, if I increase the datarate of the connection between server1 and router3, this error may pop up later, while if I decrease the datarate the error may pop up sooner.
The same is true if I change, in the .ini file, the queue.packetCapacity parameter of the nodes, or the size of the replyLength parameter of some TcpBasicClientApps, or the delayTime parameter of server1, or the thinkTime parameter of some TcpBasicClientApps, or other parameters.
The aim of the simulation is to create a bottleneck in a link (I decided to create it in the link between router3 and server1) and see how the network behaves.

Thank you for your time.

[code.zip](https://github.com/inet-framework/inet/files/6122607/code.zip)

**Comment by dstolpmann** [comment]
Hi,

we get the same error message in our simulation model. We found out that it occurs when configuring the `TcpBasicClientApp` with a high `numRequestsPerSession` or a `requestLength` that is larger than the maximum segment size (MSS) of TCP. For high `numRequestsPerSession`, the error is not always reproducible for us, but a `requestLength` that is larger than 1 MSS reliably triggers the error message.

For your model, we were able to reproduce the error for 7 or more requests per session. For smaller values, the simulation runs fine for a simulation time of 1200s.

Best regards,
Daniel
